tests/test_comp_with_appdirs.py

```diff
@@ -28,15 +37,33 @@ def test_has_backward_compatible_class() -> None:
         "app_name_author_version",
     ],
 )
-def test_compatibility(params: Dict[str, Any], func: str) -> None:
-    if sys.platform == "darwin":
+@pytest.mark.parametrize("system", ("darwin", "unix", "win32"))
+def test_compatibility(
```

Comment:
I don't agree we need to touch this test in any shape and form.

Reply:
I can revert that specific change and add it to a different PR. However, the test does need to be changed so that it doesn't fail on known compat breaks: i.e. when the supported subset of XDG_* is in `os.environ`.

Reply:
In the previous format the test does not validate compatibility with XDG on :thinking: